### ISSUE TITLE
notebook2script fails on notebooks saved by VS Code native editor

### ISSUE THREAD

**Issue by sreiz1** [fastai/nbdev#505]
Since the recent VS Code update I've started using the native notebook editor with VS Code keybindings etc. It's great, but apparently changes the notebook saved format a bit (I noticed that metadata for the notebook is now saved at the end).

This appears to confuse nbdev notebook2script - it still runs without errors, but the created python files are mostly empty, skipping most or all #export cells. (notebook2script worked fine before with the old jupyter notebook editor on the same notebooks).

This is with fastai 2.4 and nbdev 1.1.14. NB The instructions in the contributing guidelines for show_install don't work anymore.

**Comment by HDavidEspinosa** [comment]
I am having the same issue. The worst part is if one installs the new Pylance release on VSCode, it updates the notebook editor and breaks the script. 

As far as I can tell, the new native editor changes the cell data of notebooks.

I noticed that the cell structure used to be:
"cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# default_exp rpl_summarise"
   ]
  },

But after editing the cell with VSCode's new notebook editor, it is:

 {
   "cell_type": "code",
   "execution_count": 1,
   "source": [
    "# default_exp rpl_summarise"
   ],
   "outputs": [],
   "metadata": {}
  },

Note that "outputs" and "metadata" are now after "source", not before.  ALso, their order is now inverted. 

I re-runned the script after manually reorganising the elements, and it seems to correct the problem in the case of the file's first, cell. But the same solution did not work with a cell in the middle of the document, past other cells that have been edited.

**Comment by HDavidEspinosa** [comment]
Here is a nbdev project that I created using VSCode. As anyone can see, it does not build the core module. 

This might help reproduce the problem. I looked at the 00_export.ipynb of nbdev as hard as I could, but I couldn't hack it.

I have the filling that something in the new notebook structure may be braking the regex generated by `_mk_flag_re` but I am not sure.
[vsc_test.zip](https://github.com/fastai/nbdev/files/6854766/vsc_test.zip)
